### ISSUE TITLE
Settings.AudioFormat change causes NodeParameter to fail

### ISSUE THREAD

**Issue by bdrobert** [AudioKit/Cookbook#38]
I am attempting to modify the default AudioKit sample rate via Settings and am encountering an issue where the Delay Node Parameter property wrappers no longer update the AUParameters of the underlying AVAudioUnit. I have reproduced this issue in the Cookbook project Delay recipe with only the following modifications to the AppDelegate:

`func application(_ application: UIApplication,
                     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.

        #if os(iOS)
        do {
            /// The only modification to the Cookbook project
            Settings.audioFormat = AVAudioFormat(standardFormatWithSampleRate: 48_000, channels: 2)!
            
            Settings.bufferLength = .short
            try AVAudioSession.sharedInstance().setPreferredIOBufferDuration(Settings.bufferLength.duration)
            try AVAudioSession.sharedInstance().setCategory(.playAndRecord,
                                                            options: [.defaultToSpeaker, .mixWithOthers])
            try AVAudioSession.sharedInstance().setActive(true)
        } catch let err {
            print(err)
        }
        #endif

        return true
    }`

After some debugging I have found that the Delay.time and Delay.feedback do not update but the Delay.dryWetMix does update. I've debugged the Property wrapper and NodeParameters and it appears their reference no longer matches the AUParameter reference returned in the `NodeParameter.avAudioUnit.auAudioUnit.parameterTree!.allParameters`.

**Is my approach to modifying the default sample rate appropriate or am I missing something?**

**Comment by ryanfrancesconi** [comment]
I'm not super current with how AK is setup at the moment, but you probably still need to rebuild your graph after a sample rate change. You can debug by printing the `engine.debugDescription` and examine the sample rates.

**Comment by bdrobert** [comment]
Hi, I believe I am changing the sample rate before building the graph or starting the AudioEngine.  In the Cookbook project each recipe (View) manages its own AudioEngine.

**Comment by ryanfrancesconi** [comment]
Oh right - this is the cookbook.

**Comment by emurray2** [comment]
@bdrobert I tried replacing lines 13-23 of `AppDelegate.swift` in Cookbook with your code above, and the parameters worked for me. I think this may have been addressed in b25b77779425fc7271a09307606f11f3fea722cc . It's actually an issue on Apple's part, but we're filing a bug report for it.

In the meantime, try testing your code again with the [AudioKit/Cookbook develop](https://github.com/AudioKit/Cookbook/tree/develop) branch.
